pkg/review/job.go

```diff
@@ -81,61 +75,78 @@ func ExecuteJob(ctx context.Context, cfg *Config) error {
 	}
 
 	// Step 2: Get review status information for each commit.
-	// Fan out the work of processing all of the commits that were found
-	for _, commit := range commits {
-		if err := commitPool.Do(ctx, func() (*CommitReviewStatus, error) {
-			result := processCommit(ctx, *commit, gitHubClient)
-			return result, nil
-		}); err != nil {
-			return fmt.Errorf("failed to submit job to worker pool: %w", err)
-		}
-	}
-	// When all of the workers are complete, extract the result values
-	results, err := commitPool.Done(ctx)
+	commitReviewStatuses, err := pooledTransform(ctx, commits,
+		func(commit *Commit) (*CommitReviewStatus, error) {
+			return processCommit(ctx, *commit, gitHubClient), nil
```

Comment:
Is there a reason to have the processCommit func deference the commit var? Can we pass the pointer instead?

Reply:
It's  a holdover from when those functions were written to use apache beam. Changed to use a pointer instead.